chapters/mvc-basics/use-dependency-injection.md

```diff
@@ -28,13 +28,13 @@ Since `ITodoItemService` is in the `Services` namespace, you'll also need to add
 using AspNetCoreTodo.Services;
 ```
 
-The first line of the class declares a private variable to hold a reference to the `ITodoItemService`. This variable lets you use the service from the `Index` action method later.
+The first line of the class declares a private variable to hold a reference to the `ITodoItemService`. This variable lets you use the service from the `Index` action method later (you'll see how in a minute).
 
-The `public TodoController(ITodoItemService todoItemService)` line defines a **constructor** for the class. The constructor runs only once when the class is first initialized, and you've now declared that the `TodoController` class will need to be given an `ITodoItemService` to be initialized properly.
+The `public TodoController(ITodoItemService todoItemService)` line defines a **constructor** for the class. The constructor is a special method that is called when you want to create a new copy of the class (the `TodoController`, in this case). By adding an `ITodoItemService` parameter to the constructor, you've declared that in order to create the `TodoController`, you'll need to supply an object that matches the `ITodoItemService` interface.
```

Comment:
would change "create a new copy of the class" to "create a new instance of the class" to stick to proper OOP terms

Reply:
I agree with @welkie, though I feel like this still lacks a little clarity. Where do we define instance? Does a reader understand that a class is instantiated into specific object instances? Essentially, do they understand that an instance is an object?

Reply:
@davidshq I think this is where it becomes really important to define the scope of this book. It works well because it's condensed. We might want to avoid it spending too much time on basics (what is OOP? etc) or getting into the depths of ASP.NET Core.

While I think the best person to decide this direction is, @nbarbettini, my opinion is that readers should be expected to have a basic understanding of programming including object-oriented programming.

Reply:
I agree with @welkie.  This book must assume a basic level of understanding of OOP concepts.  There are plenty of existing resources that cover the basics of OOP.  Maybe expressing this assumption in a foreword or introduction, with links to some good reading resources would be sufficient?

Reply:
Awesome discussion, thanks. Playing devil's advocate: there are plenty of younger coders who maybe have experience with Node or React but don't know much about OOP yet. I shied away from using many OOP terms because I didn't want to get too jargony and potentially lose those folks. Do you agree/disagree? Interested in y'alls thoughts.

This is a pretty minor case, though - saying "new instance of the class" isn't likely to be a problem. 😄

Reply:
Wow that's actually a good point. I grew up with OOP, but lately I do indeed spend most of my time in the JS world. In fact, my school just transitioned to a new curriculum that ditches PHP and ASP.NET as an intro to web development and uses Node and Angular instead.

In that case, I think it would be a good idea have a section with an intro to OOP concepts, maybe with a disclaimer that it's an optional section that can be skipped if readers are already familiar with OOP.

It may also do double duty as a nice intro to C# language features too, which means there would be less need for comments in the rest of the book (for example, talking about the constructor, or a generic type), making the rest of the book flow nicer. In this case, the section would be best pitched as an "intro to OOP concepts in C#" so that both those new to OOP and new to C# would know to read it first before continuing.

Reply:
> my school just transitioned to a new curriculum that ditches PHP and ASP.NET as an intro to web development and uses Node and Angular instead

I have a feeling we'll see more and more of this! (Not in a bad way)

> I think it would be a good idea have a section with an intro to OOP concepts, maybe with a disclaimer that it's an optional section that can be skipped if readers are already familiar with OOP

I like this idea, although not quite sure where to stick it yet. Definitely early on.



> Those with programming experience who are new to web development and/or ASP.NET Core

That's the target audience in my mind, although I'm defining "with programming experience" loosely, so junior folks aren't excluded.